File: cs/Iris/SimpleExpressionElement/SubSimpleExpression.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Iris.SimpleExpressionElement
{
    public class SubSimpleExpression : SimpleExpression
    {
        public SimpleExpression exp;

        public SubSimpleExpression(SimpleExpression e)
        {
            exp = e;
        }

        public override string ToString()
        {
            return '(' + exp.ToString() + ')';
        }

        public override ObjectType getExpressionType()
        {
            return exp.getExpressionType();
        }
    }
}

File: java/Iris/src/com/rokuan/iris/expression/SubSimpleExpression.java
package com.rokuan.iris.expression;

import com.rokuan.iris.types.ObjectType;

public class SubSimpleExpression extends SimpleExpression
{
	public SimpleExpression exp;

	public SubSimpleExpression(SimpleExpression e)
	{
		exp = e;
	}

	public String ToString()
	{
		return '(' + exp.toString() + ')';
	}

	public ObjectType getExpressionType()
	{
		return exp.getExpressionType();
	}
}
